code/modules/lighting/lighting_system.dm

```diff
@@ -94,70 +97,125 @@
 
 //Tell the lighting subsystem to check() next fire
 /datum/light_source/proc/changed()
-	if(owner)
-		__x = owner.x
-		__y = owner.y
 	if(!changed)
 		changed = 1
 		SSlighting.changed_lights |= src
 
 //Remove current effect
 /datum/light_source/proc/remove_effect().
-	for(var/turf/T in effect)
-		T.update_lumcount(-effect[T])
+	for(var/turf/T in turfs_effect)
+		T.update_lumcount(-turfs_effect[T], turfs_direction[T])
+		T.affecting_lights -= src
 
-		if(T.affecting_lights && T.affecting_lights.len)
-			T.affecting_lights -= src
+		for(var/thing in RANGE_TURFS(1, T))
+			var/turf/neighbor = thing
```

Comment:
@duncathan yes it is, I am the authority here :P 
(for real though I've tested all possible loop formats byond has, I know which is best)